### PR TITLE
Routes and other network changes from feedback

### PR DESCRIPTION
**Pull request by DeepakRajendranMsft**
Changes include:
1. Add Routes and RouteTables
2. Rename Get-AzureCheckDnsAvailability to Test-AzureDnsAvailability (from feedback)
3. Minor loadbalancer bug
4. Allow RequestPath in loadbalancer to be optional (from feedback)
5. Tests

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@DeepakRajendranMsft__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@DeepakRajendranMsft There are merge conflicts, please update

**Comment by DeepakRajendranMsft** [comment]
the build seems to be stuck here:
 === TEST EXECUTION SUMMARY ===
     Microsoft.Azure.Commands.TrafficManager.Test.ScenarioTests  Total: 11, Errors: 0, Failed: 0, Skipped: 0, Time: 12.343s
  D:\workspace\powershell\packages\xunit.runner.console.2.0.0\tools\xunit.console.x86.exe ".\src\ServiceManagement\Network\Commands.Network.Test\bin\Debug\Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Test.dll" -trait "AcceptanceType=CheckIn" -html ".\src\Publish\TestResults\Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Test.html"
  xUnit.net console test runner (32-bit .NET 4.0.30319.34014)
  Copyright (C) 2015 Outercurve Foundation.
  
  Discovering: Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Test
  Discovered:  Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Test
  Starting:    Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Test
     Microsoft.WindowsAzure.Commands.ServiceManagement.Network.Test.ScenarioTests.NetworkTests.GetGatewayWhenGatewayDoesntExist [SKIP]
        Exploratory test by someone else not passing

Could anyone trigger a new build?

**Comment by DeepakRajendranMsft** [comment]
@markcowl Merged conflicts

File: src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj [hunk 1]
```diff
@@ -81,10 +81,7 @@
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Compute.7.0.0-preview\lib\net40\Microsoft.Azure.Management.Compute.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.2.0.8-preview\lib\net40\Microsoft.Azure.Management.Network.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
+    <Reference Include="Microsoft.Azure.Management.Network">      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.2.0.8-preview\lib\net40\Microsoft.Azure.Management.Network.dll</HintPath>    </Reference>
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj]: Please revert this change

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 2]
```diff
@@ -73,9 +73,7 @@
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Authorization.0.19.2-preview\lib\net40\Microsoft.Azure.Management.Authorization.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Azure.Management.Network, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.2.0.8-preview\lib\net40\Microsoft.Azure.Management.Network.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.2.0.8-preview\lib\net40\Microsoft.Azure.Management.Network.dll</HintPath>    </Reference>
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: Not sure what hapopened here, but again, there should not be any change here

**Comment by markcowl** [comment]
@DeepakRajendranMsft A couple of issues with csproj references to correct - otherwise looks good.  Please correct and I'll merge

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj (hunk 2 above)
Comment (DeepakRajendranMsft) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: fixed

**Comment by DeepakRajendranMsft** [comment]
@markcowl reverted the csproj changes

**Comment by markcowl** [comment]
@azuresdkci retest this please
